src/ui/navigation/container/bottomBar.tsx

```diff
@@ -0,0 +1,231 @@
+const VISIBLE_ORIGINAL_BAR_HEIGHT = 80
+const SCREEN_PROPORTION = width / ORIG_BAR_WIDTH
+// Scaled down height of the original visible bar
+const VISIBLE_BAR_HEIGHT = SCREEN_PROPORTION * VISIBLE_ORIGINAL_BAR_HEIGHT
+// Scaled down height of the whole original bar
+const BAR_HEIGHT = SCREEN_PROPORTION * ORIG_BAR_HEIGHT
+// Additional "SafeView" for extra large devices e.g. iPhone 11 Pro Max
+const BAR_EXTRA_SAFE_HEIGHT = 30
+
+/* Calculation of button size and position proportionally to the original bar */
+// Ratio of the original button size relative to the original bar width
+const BUTTON_SIZE_MODIFIER = 0.175
+const BUTTON_SIZE = BUTTON_SIZE_MODIFIER * width
+// Original distance from the center of the button to the top of the bar (vertical offset)
+const ORIGINAL_VERTICAL_OFFSET = 16
+const VERTICAL_OFFSET =
+  BUTTON_SIZE / 2 - width * (ORIGINAL_VERTICAL_OFFSET / ORIG_BAR_WIDTH)
+
+const styles = StyleSheet.create({
+  wrapper: {
+    position: 'absolute',
+    width: '100%',
+    bottom: VISIBLE_BAR_HEIGHT - BAR_HEIGHT - BAR_EXTRA_SAFE_HEIGHT,
+    height: BAR_HEIGHT + BAR_EXTRA_SAFE_HEIGHT,
+  },
+  buttonWrapper: {
+    position: 'absolute',
+    width: '100%',
+    top: 0,
+    bottom: 0,
+    height: VISIBLE_BAR_HEIGHT,
+    zIndex: 2,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingHorizontal: 10,
+  },
+  safeView: {
+    width: '100%',
+    height: BAR_EXTRA_SAFE_HEIGHT,
+    backgroundColor: '#0B030D',
+    marginTop: -4,
+  },
+})
+
+const AnimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView)
+
+interface Props
+  extends BottomTabBarProps,
+    ReturnType<typeof mapDispatchToProps> {
+  onTabPress: (scene: TabScene) => null
+  onTabLongPress: (scene: TabScene) => null
+  getTestID: (scene: TabScene) => string
+  getAccessibilityLabel: (scene: TabScene) => string
+  getAccessibilityRole: (scene: TabScene) => string
+  getAccessibilityStates: (scene: TabScene) => string[]
+}
+
+const BottomBarContainer = (props: Props) => {
+  const {
+    navigation,
+    renderIcon,
+    onTabPress,
+    getLabelText,
+    getTestID,
+    getAccessibilityLabel,
+    getAccessibilityRole,
+    getAccessibilityStates,
+    navigateInteraction,
+    safeAreaInset,
+    activeTintColor,
+    inactiveTintColor,
+  } = props
+  const { routes, index } = navigation.state
+  const colors = { activeTintColor, inactiveTintColor }
+
+  const [AnimatedHiding] = useState(new Animated.Value(0))
+
+  useEffect(() => {
+    AnimatedHiding.setValue(0)
+
+    if (navigation) {
+      const blurListener = navigation.addListener('willBlur', animateHiding)
+      const focusListener = navigation.addListener('didFocus', animateAppear)
+
+      const isIOS = Platform.OS === 'ios'
+
+      let keyboardShowListener: EmitterSubscription,
+        keyboardHideListener: EmitterSubscription
+
+      if (isIOS) {
+        keyboardShowListener = Keyboard.addListener(
+          'keyboardWillShow',
```

Comment:
I guess this can be simplified by by using `isIOS ? 'keyboardWillShow' : 'keyboardDidShow'`